src/gui/mzroll/spectrawidget.cpp

```diff
@@ -615,30 +617,35 @@ void SpectraWidget::drawAnnotations() {
 
 void SpectraWidget::drawGraph()
 {
+    clearGraph();
+    
     if (_currentScan == NULL) return;
 
     qDebug() << "showSpectra() mzrange= " << _minX << "-" << _maxX;
-    clearGraph();
 
     QColor sampleColor(Qt::black);
     if (_currentScan->sample) {
         mzSample* sample = _currentScan->sample;
         sampleColor = QColor::fromRgbF( sample->color[0], sample->color[1], sample->color[2], 1 );
     }
 
-    setScanTitle();
     drawScan(_currentScan, sampleColor);
     drawMzLabels(_currentScan);
     drawAnnotations();
 
-    // overlay spectra
-    if ( fabs(_spectralHit.precursorMz - _currentScan->precursorMz) < 0.5 ) {
-        drawSpectralHit(_spectralHit);
-        if (_currentGroup) setGroupTitle();
-    } else {
-        //TODO: either remove the check or inform user on the UI in case of failure
-       qDebug() << " overlaySpectra() skipped: " << _spectralHit.precursorMz << " " << _currentScan->precursorMz;
-    }
+    if (_spectralHit.mzList.size() > 0) {
+        setGroupTitle();
+        if (fabs(_spectralHit.precursorMz - _currentScan->precursorMz) < 0.5)
```

Comment:
Comparing a float against a double

Reply:
I'll use the almostEqual method from mzUtils for this. Thanks for pointing out!